core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/InternalTerms.java

```diff
@@ -226,7 +225,15 @@ public InternalAggregation doReduce(List<InternalAggregation> aggregations, Redu
             if (terms.getBucketsInternal().size() < getShardSize() || InternalOrder.isTermOrder(order)) {
                 thisAggDocCountError = 0;
             } else if (InternalOrder.isCountDesc(this.order)) {
-                thisAggDocCountError = terms.getBucketsInternal().get(terms.getBucketsInternal().size() - 1).docCount;
+                if (terms.getDocCountError() > 0) {
+                    // If there is an existing docCountError for this agg then
+                    // use this as the error for this aggregation
+                    thisAggDocCountError = terms.getDocCountError();
```

Comment:
Would `newAggDocCountError` be clearer here since we are working out the error for the new reduced agg?